client.go

```diff
@@ -188,10 +188,11 @@ func (c *Client) callAPI(request *http.Request) (*http.Response, error) {
 
 	for backoffIdx <= c.options.maxRetries {
 		res, err = c.do(request)
-		if err != nil && backoffIdx == c.options.maxRetries {
-			break
-		}
-		if err != nil {
+		if err != nil || res.StatusCode < 200 || res.StatusCode >= 300 {
+			if backoffIdx == c.options.maxRetries {
+				err = errors.New("non 2xx status code")
+				break
+			}
```

Comment:
This is the most important part of the PR. This will enable retries for requests with non-2xx status codes